File: webportal-client/src/types/property.ts
export interface Property {
  id: number;
  title: string;
  address: string;
  city: string;
  suburb?: string;
  price: number;
  listingType: 'Sale' | 'Rent';
  propertyType?: string;
  bedrooms: number;
  bathrooms: number;
  carSpots: number;
  floorArea?: number;
  yearBuilt?: number;
  description: string;
  imageURLs: string[];
  isFeatured?: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface PaginatedResponse<T> {
  items: T[];
  totalCount: number;
  pageNumber: number;
  totalPages: number;
  hasPreviousPage: boolean;
  hasNextPage: boolean;
}

export interface PropertyQueryParams {
  // Filter parameters
  priceFrom?: number;
  priceTo?: number;
  bedrooms?: number;
  city?: string;
  listingType?: 'Sale' | 'Rent';
  
  // Pagination parameters
  pageNumber?: number;
  pageSize?: number;
  sortBy?: string;
  sortOrder?: 'asc' | 'desc';
}

export const CITIES = [
  'Sydney',
  'Melbourne',
  'Brisbane',
  'Perth',
  'Adelaide'
] as const;

export const BEDROOM_OPTIONS = [
  { value: 1, label: '1+ Bedroom' },
  { value: 2, label: '2+ Bedrooms' },
  { value: 3, label: '3+ Bedrooms' },
  { value: 4, label: '4+ Bedrooms' },
  { value: 5, label: '5+ Bedrooms' },
];

export const LISTING_TYPES = [
  { value: 'Sale', label: 'For Sale' },
  { value: 'Rent', label: 'For Rent' },
];

File: WebPortal.API/Models/Property.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Collections.Generic;

namespace WebPortal.API.Models;

public class Property
{
    [Key]
    public int Id { get; set; }
    
    [Required]
    [StringLength(200)]
    public string Title { get; set; }
    
    [Required]
    [StringLength(500)]
    public string Address { get; set; }
    
    [Required]
    [StringLength(100)]
    public string City { get; set; }
    
    [StringLength(100)]
    public string Suburb { get; set; }
    
    [StringLength(50)]
    public string PropertyType { get; set; } // e.g., Apartment, House, Townhouse
    
    [Required]
    [Column(TypeName = "decimal(18,2)")]
    public decimal Price { get; set; }
    
    [Required]
    [StringLength(50)]
    public string ListingType { get; set; } // "Sale" or "Rent"
    
    [Required]
    public int Bedrooms { get; set; }
    
    [Required]
    public int Bathrooms { get; set; }
    
    [Required]
    public int CarSpots { get; set; }
    
    [StringLength(4000)]
    public string Description { get; set; }
    
    public List<string> ImageURLs { get; set; } = new();
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
    
    public bool IsFeatured { get; set; } = false;
    public int? FloorArea { get; set; } // in square meters
    public int? YearBuilt { get; set; }
    
    // Navigation property for users who favorited this property
    public virtual ICollection<Favorite> FavoritedBy { get; set; } = new List<Favorite>();
}
